site/app/models/User.php

```diff
@@ -461,10 +454,19 @@ public function setGradingRegistrationSections($sections) {
         }
     }
 
-    public function getAnonId() {
-        if ($this->anon_id === null) {
+    /**
+     * Get gradeable-specific anon_id of a user
+     * @param string $g_id
+     */
+    public function getAnonId($g_id) {
+        if ($g_id === "") {
+            return "";
+        }
+        $anon_id = $this->core->getQueries()->getAnonId($this->id, $g_id);
+        $anon_id = empty($anon_id) ? null : $anon_id[$this->getId()];
+        if ($anon_id === null) {
             $alpha = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890";
-            $anon_ids = $this->core->getQueries()->getAllAnonIds();
+            $anon_ids = $this->core->getQueries()->getAllAnonIdsByGradeable();
```

Comment:
Why not pass $g_id into this function?

Reply:
If `$g_id` isn't passed then all `anon_ids` in the table are returned. I intended to keep the `anon_ids` unique across gradeables, but thinking about it again I don't think it is necessary, it's just increasing the number of ids to loop over and thus increasing the computation time, so it has now been modified to send `$g_id` too.